process/block/metablock.go

```diff
@@ -44,43 +39,29 @@ type metaProcessor struct {
 }
 
 // NewMetaProcessor creates a new metaProcessor object
-func NewMetaProcessor(
-	core serviceContainer.Core,
-	accounts state.AccountsAdapter,
-	dataPool dataRetriever.MetaPoolsHolder,
-	forkDetector process.ForkDetector,
-	shardCoordinator sharding.Coordinator,
-	nodesCoordinator sharding.NodesCoordinator,
-	specialAddressHandler process.SpecialAddressHandler,
-	hasher hashing.Hasher,
-	marshalizer marshal.Marshalizer,
-	store dataRetriever.StorageService,
-	startHeaders map[uint32]data.HeaderHandler,
-	requestHandler process.RequestHandler,
-	uint64Converter typeConverters.Uint64ByteSliceConverter,
-) (*metaProcessor, error) {
+func NewMetaProcessor(arguments ArgMetaProcessor) (*metaProcessor, error) {
 
 	err := checkProcessorNilParameters(
```

Comment:
you can change this function to send the entire structure to it and test the parameters used in shardblock and metablock.

Reply:
Done.